src/test/java/com/yanna/stepanova/service/ShoppingCartServiceTest.java

```diff
@@ -0,0 +1,83 @@
+package com.yanna.stepanova.service;
+
+import com.yanna.stepanova.dto.cartitem.CartItemDto;
+import com.yanna.stepanova.dto.shoppingcart.ShoppingCartDto;
+import com.yanna.stepanova.mapper.ShoppingCartMapper;
+import com.yanna.stepanova.model.ShoppingCart;
+import com.yanna.stepanova.repository.shoppingcart.ShoppingCartRepository;
+import com.yanna.stepanova.service.impl.ShoppingCartServiceImpl;
+import jakarta.persistence.EntityNotFoundException;
+import java.util.Optional;
+import java.util.Set;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class ShoppingCartServiceTest {
+    @Mock
+    private ShoppingCartRepository shoppingCartRepo;
+    @Mock
+    private ShoppingCartMapper shoppingCartMapper;
+    @InjectMocks
+    private ShoppingCartServiceImpl shoppingCartService;
+
+    @Test
+    @DisplayName("""
+            Get shopping cart by user id""")
+    public void getShopCart_WithValidId_ReturnShoppingCart() {
+        //given
+        Long userId = 5L;
+        ShoppingCart expected = new ShoppingCart();
+        expected.setId(userId);
+
+        Mockito.when(shoppingCartRepo.findById(userId)).thenReturn(Optional.of(expected));
+        //when
+        ShoppingCart actual = shoppingCartService.getShopCart(userId);
+        //then
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("""
+            Get exception for non-existing user id""")
+    public void getShopCart_NonExistingId_ReturnException() {
+        // given
+        Long userId = 2L;
+        String expected = "Can't find shopping cart by id:" + userId;
+        Mockito.when(shoppingCartRepo.findById(userId)).thenReturn(Optional.empty());
+        // when
+        try {
```

Comment:
```suggestion

        // when
        try {
```
better leave an empty line between given/when/then blocks